xwiki-rendering-macros/xwiki-rendering-macro-content/src/main/java/org/xwiki/rendering/internal/macro/content/ContentMacro.java

```diff
@@ -87,7 +90,18 @@ public boolean supportsInlineMode()
         throws MacroExecutionException
     {
         try {
-            return getSyntaxParser(parameters.getSyntax()).parse(new StringReader(content)).getChildren();
+            List<Block> blocks = getSyntaxParser(parameters.getSyntax()).parse(new StringReader(content)).getChildren();
+
+            // if the result is only one block, we put the syntax metadata in it
+            if (blocks.size() == 1) {
+                blocks.get(0).setParameter(MetaData.SYNTAX, parameters.getSyntax().toIdString());
```

Comment:
You are mixing two different things. Block parameters have nothing to do with MetaData, you should check if the Block is a MetaDataBlock.

Reply:
OK actually as the Javadoc of `Block#getParameter` was stating: 
> A Parameter is a generic key/value which can be used to add metadata to a block.

I assumed that it could avoid us to introduce a new MetadataBlock. But now I imagine it's not the same kind of metadata: parameters are considered only for "relevant" parameter of a block, such as the target for a link or an image. Is that right? (Sorry, still trying to remove some ambiguities I might have)

Reply:
Yes we should be more careful in the javadoc and try to avoid using "metadata" words for things not related to the actual Metadata class even if it's correct in English.

Reply:
Thanks for the explanation. Edited the PR.